adagucserverEC/CDBAdapterPostgreSQL.cpp

```diff
@@ -356,181 +374,104 @@ CDBStore::Store *CDBAdapterPostgreSQL::getFilesAndIndicesForDimensions(CDataSour
-                  }
-                }
-              }
-
-              delete dataType;
-              dataType = NULL;
-
-            } catch (int e) {
-              if ((CServerParams::checkDataRestriction() & SHOW_QUERYINFO) == false) query.copy("hidden");
-              CDBError("Unable to determine column type: '%s'", dataTypeQuery.c_str());
-              return NULL;
-            }
-
-            if (l > 0) subQuery.concat("and ");
-            if (sDims->count == 1) {
-
-              if (!CServerParams::checkTimeFormat(sDims[l])) timeValidationError = true;
-
-              if (isRealType == false) {
-                subQuery.printconcat("%s = '%s' ", netCDFDimName.c_str(), sDims[l].c_str());
-              }
-
-              // This query gets the closest value from the table.
-              if (isRealType) {
-                subQuery.printconcat("abs(%s - %s) = (select min(abs(%s - %s)) from %s)", sDims[l].c_str(), netCDFDimName.c_str(), sDims[l].c_str(), netCDFDimName.c_str(), tableName.c_str());
-              }
-            }
-
-            // TODO Currently only start/stop is supported, start/stop/resolution is not supported yet.
-            if (sDims->count >= 2) {
-              if (l == 0) {
-                if (!CServerParams::checkTimeFormat(sDims[l])) timeValidationError = true;
-                // Get closest lowest value to this requested one, or if request value is way below get earliest value:
-                subQuery.printconcat("%s >= (select max(%s) from %s where %s <= '%s' or %s = (select min(%s) from %s)) ", netCDFDimName.c_str(), netCDFDimName.c_str(), tableName.c_str(),
-                                     netCDFDimName.c_str(), sDims[l].c_str(), netCDFDimName.c_str(), netCDFDimName.c_str(), tableName.c_str());
-                // subQuery.printconcat("%s >= '%s' ",netCDFDimName.c_str(),sDims[l].c_str());
-              }
-              if (l == 1) {
-                if (!CServerParams::checkTimeFormat(sDims[l])) timeValidationError = true;
-                subQuery.printconcat("%s <= '%s' ", netCDFDimName.c_str(), sDims[l].c_str());
-              }
-            }
-          }
-        }
-        delete[] sDims;
-      }
-      delete[] cDims;
-    }
-    if (i == 0) {
-      if (numQueriesAdded == 0) {
-        subQuery.printconcat("where ");
-      } else {
-        subQuery.printconcat("and ");
+    for (int i = 0; i < requestedDimVals->count; ++i) {
+      if (requestedDimVals[i].equals("*")) continue;
+
+      if (i != 0) {
+        whereStatement.printconcat(" OR ");
       }
-      if (dataSource->queryBBOX) {
 
-        subQuery.printconcat("adaguctilinglevel = %d and minx >= %f and maxx <= %f and miny >= %f and maxy <= %f ", dataSource->queryLevel, dataSource->nativeViewPortBBOX[0],
-                             dataSource->nativeViewPortBBOX[2], dataSource->nativeViewPortBBOX[1], dataSource->nativeViewPortBBOX[3]);
+      CT::string *dimVals = requestedDimVals[i].splitToArray("/");
```

Comment:
BTW, I think we should think of CT::string split options. splitToStack will create a std::vector<CT::string>.